include/zephyr/device.h

```diff
@@ -1079,6 +1079,41 @@ __syscall int device_init(const struct device *dev);
 
 DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)
 
+/**
+ * @brief Wrapper macro for declaring device API structs inside iterable sections.
+ *
+ * @param _type The device API type.
+ * @param _name The API instance name.
+ */
+#define DEVICE_API(_type, _name) const STRUCT_SECTION_ITERABLE(_type, _name)
```

Comment:
Should we have a prefix in the section name (using `STRUCT_SECTION_ITERABLE_NAMED`)? I'm thinking that there's going to potentially be quite a few of these sections, it'd be nice to have them on a common prefix so they are grouped together in elf analysis tools (readelf or gdb info files).